File: python/dc_restaurants.py
# Import the modules, endpoint, headers and api_key
import requests
import json
import sys


def dc_restaurants_average(API_KEY, ENDPOINT, HEADERS):

    offset = -50
    dc_restaurants_rating = 0
    counter = 0

    # BUSINESS SEARCH PARAMETERS
    while( counter <= 6600):
        if offset < 950:
            offset += 50
        else:
            break
            
        PARAMETERS_DC = {
                    "categories":[{"alias": "fastfood", "title": "Fast Food"}],
                    'limit': 50,
                    'offset': offset,
                    'radius': 40000,
                    'location': 'Washington, DC'}

        # Make a request to the Yelp API
        response_dc = requests.get(url = ENDPOINT,
                                params = PARAMETERS_DC,
                                headers = HEADERS)

        # Conver the JSON String and extract the restaurants
        dc_data = response_dc.json()
        dc_restaurants = dc_data['businesses']

        #print the response for McDonald's
        aux = 0
        for restaurant in dc_restaurants:
            if restaurant['location']['state'] == 'DC' and restaurant['location']['city'] == 'Washington, DC':
                dc_restaurants_rating += restaurant['rating']
                counter += 1
    print("Average DC restaurants rating: {}".format(round((dc_restaurants_rating / counter), 2)))

sys.modules[__name__] = dc_restaurants_average

File: nodejs/DcRestaurants.js
const fetch = require('node-fetch');

async function dcRestaurants(apiKey, endpoint, headers) {

    let sumOfRates = 0;
    let totalRestaurants = 0;

    for (let offset = 0; offset < 950; offset += 50) {

        const params = {
            "categories": [{ "alias": "fastfood", "title": "Fast Food" }],
            'limit': 50,
            'offset': offset,
            'radius': 10000,
            'location': 'Washington, DC'
        }

        try {
            let baseUrl = new URL(endpoint);
            Object.keys(params).forEach(key => baseUrl.searchParams.append(key, params[key]));

            const response = await fetch(baseUrl, { headers: headers });
            const data = await response.json();
            const restaurants = data.businesses;

            for (let restaurant of restaurants) {
                sumOfRates += restaurant.rating;
                totalRestaurants += 1;
            }
        }
        catch (e) {
            console.log(e);
        }

    }

    const average = sumOfRates / totalRestaurants;
    console.log("DC average:", Math.round((average + Number.EPSILON) * 100) / 100);
}

module.exports = dcRestaurants;